Automattic/pocket-casts-android
# Bookmarks - Sort type updates

## Description

This 
- Converts bookmarks sort type to user settings
- Saves bookmarks sort type separately for player and episode screens in addition to that for podcast details screen

Part of https://github.com/Automattic/pocket-casts-android/issues/1307

## Testing Instructions

1. Open the app as a Patron user
2. Add a few bookmarks for a podcast (for different episodes)
3. Add a few bookmarks for a user episode
4. Set bookmark sort order on `Podcast Details` screen to `Episode`
5. Set bookmark sort order on `Episode Details` or on `(User Episodes) Files` screen to `Timestamp`
6. Set bookmark sort order on `Player` screen to `Oldest to newest`
7. ✅ Go back to `Podcast Details` screen and notice that bookmark sort order is  `Episode` and bookmarks are sorted accordingly
9. ✅ Go back to `Episode Details` screen or `(User Episodes) Files` screen and notice that bookmark sort order is `Timestamp` and bookmarks are sorted accordingly
10. ✅ Go back to `Player` screen and notice that bookmark sort order is `Oldest to newest`  and bookmarks are sorted accordingly
11. Close the app
12. Reopen and notice that sort orders are retained


Note: This commit (https://github.com/Automattic/pocket-casts-android/pull/1321/commits/2f23fa95df572cfc918a9e99dae0839871f1305d) possibly fixes the below issue as I could see some discrepancies with the remembered date values on scroll bookmarks list.

> The oldest/newest sort options are reversed
> When sorting Oldest to Newest displays the newest bookmark first, and newest to oldest shows the oldest bookmark first.

Here's how see it on changing sort orders:

https://github.com/Automattic/pocket-casts-android/assets/1405144/275aee3e-2574-49d3-8f09-75ac80c7dad8


This commit (https://github.com/Automattic/pocket-casts-android/pull/1321/commits/e48dacf2ad3fcc660ac800472771597984012918) updates bookmark row key.

## Checklist
 
#### I have tested any UI changes...
